core/templates/dev/head/pages/exploration-editor-page/exploration-editor-page.controller.ts

```diff
@@ -654,11 +661,14 @@ angular.module('oppia').directive('explorationEditorPage', [
           // Remove save from tutorial if user does not has edit rights for
           // exploration since in that case Save Draft button will not be
           // visible on the create page.
-          if (!GLOBALS.can_edit) {
-            var index = ctrl.EDITOR_TUTORIAL_OPTIONS.indexOf(
-              saveButtonTutorialElement);
-            ctrl.EDITOR_TUTORIAL_OPTIONS.splice(index, 1);
-          }
+          UserExplorationPermissionsService.getPermissionsAsync()
+            .then(function(permissions) {
+              if (permissions.can_edit) {
+                var index = ctrl.EDITOR_TUTORIAL_OPTIONS.indexOf(
+                  saveButtonTutorialElement);
+                ctrl.EDITOR_TUTORIAL_OPTIONS.splice(index, 1);
+              }
+            });
```

Comment:
Can you move this together with the code on line 437.

Reply:
The code on line 437 is executed after the exploration page has been initialized.
This is done without initializing the exploration page.